datafusion/sql/src/planner.rs

```diff
@@ -85,6 +85,10 @@ pub trait ContextProvider {
 
     /// Get configuration options
     fn options(&self) -> &ConfigOptions;
+
+    fn udfs(&self) -> HashMap<String, Arc<ScalarUDF>>;
```

Comment:
This sort of exposes details of the implementation (e.g. a `HashMap`)

What about potentially just returning the names following the model of CatalogProvider or SchemaProvider: https://docs.rs/datafusion/latest/datafusion/catalog/schema/trait.SchemaProvider.html#tymethod.table_names


Something like

```suggestion
    /// returns all udf names
    fn udf_names(&self) -> Vec<&str>
```

(or maybe Strings)

Reply:
Nice! I love consistency! Will do it.

Reply:
Do you want this function to return a `Vec<&str>` or `Vec<String>`, the suggested change uses `Vec<&str>`, but the `table_names()` function in the referenced link uses `Vec<String>`